Cryptic error when test node name start with `run_tests_ `

```
CMake Error at /opt/ros/kinetic/share/catkin/cmake/test/tests.cmake:114 (add_dependencies):
  Cannot add target-level dependencies to non-existent target
  "_run_tests_mynode".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.
Call Stack (most recent call first):
```
https://answers.ros.org/question/304047/cannot-add-target-level-dependencies-to-non-existent-target/

--- a/cmake/test/tests.cmake
+++ b/cmake/test/tests.cmake
@@ -97,6 +97,11 @@ function(catkin_run_tests_target type name xunit_filename)
     message(FATAL_ERROR "catkin_run_tests_target() called with unused arguments: ${_testing_UNPARSED_ARGUMENTS}")
   endif()
 
+  # Friendly error message for ros/catkin#961
+  if(TARGET run_tests_${PROJECT_NAME} AND NOT TARGET _run_tests_${PROJECT_NAME})
+    message(FATAL_ERROR "catkin_run_tests_target() needs to create a target called `run_tests_${PROJECT_NAME}`, but it already exists. Please rename the existing `run_tests_${PROJECT_NAME}` target/executable/library to something else.")
+  endif()
+
   # create meta target to trigger all tests of a project
   if(NOT TARGET run_tests_${PROJECT_NAME})
     add_custom_target(run_tests_${PROJECT_NAME})

